controller_manager:
  ros__parameters:
    update_rate: 100 # Hz

    omron_forward_controller:
      type: omron/OmronController

    omron_state_broadcaster:
      type: omron/OmronStateBroadcaster

    omron_fake_position_controller:
      type: omron/OmronFakePositionController

omron_fake_position_controller:
  ros__parameters:
    feedback:
      use_open_loop: true
      kp: 0.0
    cmd_vel_topic: "/omron/cmd_vel"
    tf:
      use_tf: true
      to: "base_link"
      from: "map"
      odom: "odom"
    interfaces:
      velocity:
        prefix: "omron_base/velocity"
        command:
          - "linear/velocity"
          - "angular/velocity"
        states:
          - "linear/velocity"
          - "angular/velocity"
      pose:
        prefix: "omron_base/pose"
        command:
        - "linear/x"
        - "linear/y"
        - "angular/z"
        states:
        - "linear/x"
        - "linear/y"
        - "angular/z"
      reference:
        prefix: "omron_base/reference"
        command:
          - "linear/x"
#          - "linear/y"
#          - "linear/z"
#          - "angular/x"
#          - "angular/y"
          - "angular/z"

omron_state_broadcaster:
  ros__parameters:
    from_deg_to_rad: false
    tf:
      use_tf: true
      to: "base_link"
      from: "map"
      odom: "odom"
    interfaces:
      velocity:
        prefix: "omron_base/velocity"
        command:
          - "linear/velocity"
          - "angular/velocity"
        states:
          - "linear/velocity"
          - "angular/velocity"
      pose:
        prefix: "omron_base/pose"
        states:
        - "linear/x"
        - "linear/y"
        - "angular/z"
      reference:
        prefix: "omron_base/reference"
        command:
          - "linear/x"
    #          - "linear/y"
    #          - "linear/z"
    #          - "angular/x"
    #          - "angular/y"
          - "angular/z"

